apps/examples/testcase/ta_tc/ble_manager/itc/itc_blemanager_main.c

```diff
@@ -0,0 +1,865 @@
+
+	for (i = 1; i <= LOOP_SIZE; i++) {
+		ret = ble_client_connect(ctx, &conn_info);
+		TC_ASSERT_EQ_CLEANUP("ble_client_connect", ret, BLE_MANAGER_SUCCESS, ble_manager_deinit());
+		ITC_FUNC_WAIT;
+
+		gettimeofday(&start_t, NULL);
+		ret = ble_client_disconnect(ctx);
+		TC_ASSERT_EQ_CLEANUP("ble_client_disconnect", ret, BLE_MANAGER_SUCCESS, ble_manager_deinit());
+		ITC_FUNC_WAIT;
+		gettimeofday(&end_t, NULL);
+		average += diff_time(&start_t, &end_t);
+	}
+
+	ret = ble_manager_deinit();
+	TC_ASSERT_EQ("ble_manager deinit", ret, BLE_MANAGER_SUCCESS);
+	printf("\n[Average disconnect time: %.2f ms]\n", average / (LOOP_SIZE * 1000));
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @testcase         itc_blemanager_average_stoping_time
+ * @brief            check average time taken to deinit
+ * @scenario         check initialize, de-initalization of ble manager
+ * @apicovered       ble_manager_init, ble_manager_deinit
+ * @precondition     none
+ * @postcondition    none
+ */
+static void itc_blemanager_average_stoping_time(void)
+{
+	int i = 0;
+	double average = 0.0f;
+	struct timeval start_t;
+	struct timeval end_t;
+
+	ble_result_e ret = BLE_MANAGER_FAIL;
+
+	for (i = 1; i <= LOOP_SIZE; i++) {
+		ret = ble_manager_init(NULL);
+		TC_ASSERT_EQ("ble_manager_init", ret, BLE_MANAGER_SUCCESS);
+
+		gettimeofday(&start_t, NULL);
+		ret = ble_manager_deinit();
+		TC_ASSERT_EQ("ble_manager deinit", ret, BLE_MANAGER_SUCCESS);
+		gettimeofday(&end_t, NULL);
+		average += diff_time(&start_t, &end_t);
+	}
+
+	printf("\n[Average deinit time: %.2f ms]\n", average / (LOOP_SIZE * 1000));
+	TC_SUCCESS_RESULT();
+}
+
+int blemanager_itc(int argc, FAR char *argv[])
+{
+	if (testcase_state_handler(TC_START, "BLE Manager ITC") == ERROR) {
+		return ERROR;
+	}
+
+	addr_string_to_mac(argv[1]);
+
+	//init_deinit
+	itc_blemanager_init_deinit_p();
+	itc_blemanager_init_deinit_n();
+
+	//scan
+	itc_blemanager_scan_p();
+
+	//whitelist
+	itc_blemanager_whitelist_add_del_p();
+
+	//connect_disconnect
+	itc_blemanager_connect_disconnect_p();
+	itc_blemanager_connect_disconnect_n();
+
+	//connection info
+	itc_blemanager_client_con_list_p();
+	itc_blemanager_connected_dev_info_p();
+
+	//bonded device
+	itc_blemanager_bonded_device_p();
+
+	//advertisement
+	itc_blemanager_advertisement_p();
+	itc_blemanager_readvertisement_p();
+
+	//profile count
+	itc_blemanager_get_profile_count_p();
+
+	//average time
+	itc_blemanager_average_connect_time();
+	itc_blemanager_average_disconnect_time();
+	itc_blemanager_average_stoping_time();
+
+	sem_destroy(&g_func_sem);
+
+	(void)testcase_state_handler(TC_END, "BLEManager ITC");
+	return 0;
+}
+
+int itc_blemanager_main(int argc, char *argv[])
+{
+	int res = sem_init(&g_func_sem, 0, 0);
+	if (res != 0) {
+		printf("g_func_sem init failed\n");
+		return ERROR;
+	}
+
+	if (argc != 2) {
+		printf("Please provide mac addr of BLE server to run ITC against\n");
+		return ERROR;
+	}
+
+	blemanager_itc(argc, argv);
+
+	return 0;
+}
```

Comment:
Could you leave a newline at the end of the file please?

Reply:
Sure, done